### DISCUSSION TITLE
Slow neovim startup time due to opening buffers

### DISCUSSION THREAD

**Discussion by michaelmaitland** [neovim/neovim#25676] [Q&A]
### Problem

Hi All,

I noticed very slow nvim startup time. I have built nvim from source:

```
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1692716794
```

When I run `nvim --startuptime startup.log MyFile.cpp` here is what I get: https://pastebin.com/G2Kzcgrf. It looks like the problem is occuring on line 328: `10672.003  10377.402: opening buffers`. The file I am trying to open is ~3000LOC of C++ code, although opening smaller C++ files (200LOC) or even my nvim/init.lua (100LOC) also faces the same long startup time.

You can see my health check: https://pastebin.com/UZSBr5BW

I am connected through ssh, attached to a tmux session on that device, and opening the file using nvim. The file is on the device I am sshed into and that the tmux session is over. The issue still exists even if I am not using tmux.

I have tried the same init.lua file on my host machine and I do not face this problem. I am using https://github.com/nvim-lua/kickstart.nvim/blob/master/init.lua as my `.config/nvim/init.lua` file.

Any ideas on what may be going wrong when opening buffers and what I can do to remedy?

### Steps to reproduce

* ssh me@mydevice
* tmux
* Build NVIM v0.9.4 Build type: Release LuaJIT 2.1.1692716794 (using stable branch)
* Set `.config/nvim/init.lua` as https://github.com/nvim-lua/kickstart.nvim/blob/master/init.lua

### Expected behavior

It should not take that long to open buffers.

### Neovim version (nvim -v)

NVIM v0.9.4 Build type: Release LuaJIT 2.1.1692716794

### Vim (not Nvim) behaves the same?

VIM - Vi IMproved 9.0 does not have this problem

### Operating system/version

RedHat 4.18.0

### Terminal name/version

iTerm

### $TERM environment variable

screen

### Installation

build from repo

**Comment by clason** [comment]
Are you sure the treesitter parsers etc. are already installed on the target machine when running the startup log? Try disabling treesitter altogether (in fact, that should be the first step: minimize the config until the timings align, then you know the source of the problem and can investigate further).

**Comment by lucario387** [comment]
Looks oddly familiar to https://github.com/nvim-treesitter/nvim-treesitter-textobjects/issues/507

**Comment by michaelmaitland** [comment]
I did some experiments and I think that nvim-lspconfig may be having a big impact. I saw slowdown with LSP plugin but Mason, nvim-cmp, treesitter, telescope commented out.

**Comment by justinmk** [comment]
Try https://vi.stackexchange.com or https://github.com/neovim/neovim/discussions for questions about usage and configuration.

The issue tracker is for bugs and feature requests.
